tests/js/counter.spec.js

```diff
@@ -0,0 +1,148 @@
+import {
+  createNewPost,
+  enablePageDialogAccept,
+  getEditedPostContent,
+  insertBlock,
+  createURLMatcher,
+  createJSONResponse, setUpResponseMocking,
+} from '@wordpress/e2e-test-utils';
+
+import {
+  selectBlockByName,
+  openSidebarPanelWithTitle,
+  typeInInputWithLabel,
+  selectStyleByName,
+  typeInInputWithPlaceholderLabel,
+  typeInTextareaWithLabel,
+} from './e2e-tests-helpers';
+
+import 'expect-puppeteer';
+
+const COUNTER_API_URL = `https://global-petition-counter.appspot.com/counter/palmoil`;
+
+
+const COUNTER_API_MOCK_DATA = {
+  "unique_count": 1063401
+};
+
+const COUNTER_API_MOCK_RESPONSES = [
+  {
+    match:  createURLMatcher( COUNTER_API_URL ),
+    onRequestMatch: createJSONResponse( COUNTER_API_MOCK_DATA ),
+  },
+];
+
+describe( 'Counter block', () => {
+  beforeAll( async () => {
+    // This helps in overriding Wordpress dialogs preventing actions
+    await enablePageDialogAccept();
+  } );
+  beforeEach( async () => {
+    // Before running each test, go to the create post page
+    await createNewPost();
+
+    // Insert block by title
+    await insertBlock( 'Counter' );
+  } );
+
+  // This is the first test, tests starts with it().
+  it( 'is inserted into the Editor', async () => {
+    // Check if block was inserted
+    expect( await page.$( '[data-type="planet4-blocks/counter"]' ) ).not.toBeNull();
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  }, 50000 );
+
+  it( 'test the counter block with text only style', async () => {
+    await selectBlockByName( 'planet4-blocks/counter' );
+
+    // Add richtext inputs.
+    await typeInInputWithPlaceholderLabel( 'Enter title', 'Text only counter block' );
+    await typeInInputWithPlaceholderLabel( 'Enter description', 'Counter block description' );
+
+    // Select the Text only counter style
+    await openSidebarPanelWithTitle( 'Styles' );
+    await selectStyleByName( 'Text Only' );
+
+    // Select the Text only counter style
+    await openSidebarPanelWithTitle( 'Setting' );
+    await typeInInputWithLabel( 'Completed', '75' );
```

Comment:
Maybe we should integers instead of strings for the "Completed" and "Target" fields?

Reply:
The `page.keyboard.type();` function expect the text input. If we pass number it throws - “TypeError: text is not iterable”